src/lib/storage/frame_of_reference_segment.cpp

```diff
@@ -52,22 +53,30 @@ template <typename T, typename U>
 std::shared_ptr<BaseSegment> FrameOfReferenceSegment<T, U>::copy_using_allocator(
```

Comment:
We probably should add another test parameter to `SegmentsUsingAllocatorsTest` to test segments with and without null values.

Reply:
Modified the tests. Without digging too deep into the allocator tracking I am not really certain that everything is not tested. Can you take a quick look at it?